### ISSUE TITLE
UnboundLocalError: local variable 'pokemon_data' referenced before assignment

### ISSUE THREAD

**Issue by Erody** [PokemonGoF/PokemonGo-Bot#1831]
### Actual Behavior
Bot sometimes crashes with this error message: UnboundLocalError: local variable 'pokemon_data' referenced before assignment

### Other Information
OS: Ubuntu 14.04
Git Commit: 79152bc65f8194ee9eb591fa657b99423771409e

**Comment by hklcf** [comment]
same here

2f277fd7d08ff9a7ad45cf46e102a00db7ba9fdf

**Comment by camboui** [comment]
Me to : 
Traceback (most recent call last):
  File "pokecli.py", line 450, in <module>
    main()
  File "pokecli.py", line 68, in main
    bot.tick()
  File "/home/fry/PokemonGO/PokemonGo-Bot/pokemongo_bot/__init__.py", line 84, in tick
    if worker.work() == WorkerResult.RUNNING:
  File "/home/fry/PokemonGO/PokemonGo-Bot/pokemongo_bot/cell_workers/incubate_eggs_worker.py", line 29, in work
    self._hatch_eggs()
  File "/home/fry/PokemonGO/PokemonGo-Bot/pokemongo_bot/cell_workers/incubate_eggs_worker.py", line 134, in _hatch_eggs
    for pokemon in pokemon_data:
UnboundLocalError: local variable 'pokemon_data' referenced before assignment

**Comment by knobunc** [comment]
```
[08:23:24] Highest CP Pokemon: Nidoran M [CP: 413] [IV: 7/14/15] Potential: 0.8 
[08:23:24] Most Perfect Pokemon: Nidoran M [CP: 413] [IV: 7/14/15] Potential: 0.8 
Traceback (most recent call last):
  File "pokecli.py", line 464, in <module>
    main()
  File "pokecli.py", line 68, in main
    bot.tick()
  File "/home/fiji/public_html/PokemonGo-Bot/pokemongo_bot/__init__.py", line 84, in tick
    if worker.work() == WorkerResult.RUNNING:
  File "/home/fiji/public_html/PokemonGo-Bot/pokemongo_bot/cell_workers/incubate_eggs_worker.py", line 29, in work
    self._hatch_eggs()
  File "/home/fiji/public_html/PokemonGo-Bot/pokemongo_bot/cell_workers/incubate_eggs_worker.py", line 134, in _hatch_eggs
    for pokemon in pokemon_data:
UnboundLocalError: local variable 'pokemon_data' referenced before assignment
```

**Comment by reddivision** [comment]
Fix to address this is awaiting merge #1777
